### PR TITLE
Revert "Revert "Use a common FNET_Transport owned by Proton in both SceduledExecutor …""

### PR DESCRIPTION
**Pull request by baldersheim**
Reverts vespa-engine/vespa#21294

### CODE REVIEW

**Comment by baldersheim** [comment]
@geirst PR
Only last commit is new.

File: searchcore/src/vespa/searchcore/proton/server/proton_config_fetcher.cpp [hunk 1]
```diff
@@ -162,21 +166,26 @@ ProtonConfigFetcher::getGeneration() const
...
+ProtonConfigFetcher::start(FastOS_ThreadPool & threadPool)
 {
     fetchConfigs();
-    if (_threadPool.NewThread(this, nullptr) == nullptr) {
+    if (threadPool.NewThread(this, nullptr) == nullptr) {
         throw vespalib::IllegalStateException(
                 "Failed starting thread for proton config fetcher");
     }
+    lock_guard guard(_mutex);
+    _running = true;
```
Comment (toregge) [inline searchcore/src/vespa/searchcore/proton/server/proton_config_fetcher.cpp:177]: Consider setting `_running` to true before trying to start the thread and set it back to false on failure.

Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/proton_config_fetcher.cpp]: Done, also prevents starting another thread if start() is called twice.

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**